native-schema-registry/c/include/mutable_byte_array.h

```diff
@@ -0,0 +1,45 @@
+#ifndef MUTABLE_BYTE_ARRAY_H
+#define MUTABLE_BYTE_ARRAY_H
+#include <stdlib.h>
+#include "glue_schema_registry_error.h"
+
+//Integer.MAX_VALUE in Java
+//This gives ~2.1Gb limit on a record.
+#define MAX_BYTES_LIMIT 2147483647
+/**
+* A mutable byte array that allows write / updating bytes in a fixed array of size `max_len`.
+*/
+typedef struct mutable_byte_array {
```

Comment:
if you're already using aws-c-common, why not use aws_byte_cursor and aws_byte_buf?

cursor is read only slice, and byte_buf is mutable.

Reply:
Thanks for the suggestion. I have modified the `mutable_byte_array` to wrap a `aws_byte_buf` so that I don't do buffer management myself. The outer structure is still useful to have as it fits our exception handling and binding generation model.

I wanted to do the same by wrapping `aws_byte_cursor` under `read_only_byte_array` but couldn't find a method that would return a pointer to it (I might be missing something). I could only find this [method](https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/byte_buf.h#L585).

Reply:
You can return cursor.ptr in that case